lib/BridgeAbstract.php

```diff
@@ -381,6 +381,68 @@ public function detectParameters($url){
 		}
 	}
 
+	/**
+	 * Get a variety of input parameters to test. By default uses the
+	 * example/defaultValues of the bridge.
+	 *
+	 * @return array of valid testing parameters, for each context. Ex:
+	 * array(
+	 *    array('context' => 'context1', $name => $value, ...),
+	 *    array('context' => 'context1', $name => $value, ...),
+	 *    array('context' => 'context2', $name => $value, ...),
+	 *    ...
+	 *  ),
+	 *  ...
+	 * )
+	 */
+	public function getTestParameters(){
+		// Set default/exampleValue for global parameters
+		$globals = array();
```

Comment:
I am especially wary of adding more and more features to bridges, especially ones that run mutating methods.

In my mind, ideally, the testing of bridges would be handled by separate test suite classes. But I see why that would not be practical with the way the repo is currently structured so I would propose:

1. Moving the parameter generator to a separate `BridgeParametersGenerator` class since `BridgeAbstract` is already responsible for too much stuff. Then `BridgeAbstract::getTestParameters` could either use `BridgeParametersGenerator`, or, better, it would just return `array|null` and the checker would be responsible for calling `BridgeParametersGenerator` in the `null` case.

Reply:
I feel like this adds complexity for no real benefit.